website/client/js/controllers/settingsCtrl.js

```diff
@@ -287,6 +293,33 @@ habitrpg.controller('SettingsCtrl',
       return Math.floor(numberOfHourglasses);
     };
 
+    $scope.hasBackupAuthOption = function(user, checkedNetworkKey) {
+      if (user.auth.local.username) {
+        return true;
+      }
+      return SOCIAL_AUTH_NETWORKS.find(function (network) {
+        if (network.key !== checkedNetworkKey) {
+          if (user.auth.hasOwnProperty(network.key)) {
+            return user.auth[network.key].id;
+          }
+        }
+      });
+    };
+
+    $scope.hasSocialAuth = function (user) {
+      return SOCIAL_AUTH_NETWORKS.find(function (network) {
+        if (user.auth.hasOwnProperty(network.key)) {
+          return user.auth[network.key].id;
+        }
+      });
+    };
+
+    $scope.deleteSocialAuth = function (networkKey) {
+      $http("delete", ApiUrl.get() + "/api/v3/user/auth/social/"+networkKey, null, "detachedSocial");
```

Comment:
@vIiRuS So does the "detachedSocial" string get used for localization within the `$http` module?  Is there any way to pass in a pre-localized message?